mpc_controller:
    ros__parameters:
        radius: 25.0
        cy: 70.0
        wheel_base: 1.88
        max_steer_angle: 0.3
        max_steer_rate: 0.3
        max_velocity: 10.0
        max_acc: 0.8
        traj_dl: 0.2
        traj_length: 50
        timer_period: 0.05
        mpc_steps: 6
        mpc_dt: 0.15
        kcte: 1.1
        kepsi: 2.0
        kev: 0.1
        ksteer_cost: 5.0

        # radius: 25.0 # Это допустимая погрешность радуса области поворота автомобиля
        # cy: 70.0 # Это параметр, который определяет центр траектории
        # traj_length: 50 # Длина траектории движения, по которой автомобиль должен ехать
        # timer_period: 0.05 # Период работы цикла контроллера управления
        # max_velocity: 10.0 # Максимальная скорость движения автомобиня
        # max_steer_rate: 0.3 # Максимальная скорость поворота рулевого колеса
        # mpc_steps: 6  # На какое количество периодов (шагов) в будущем идёт предсказание методом MPC от текущего момента времени
        # mpc_dt: 0.15 # Этот параметр определяет временной интервал между шагами работы в методе MPC
        # kcte: 1.1 # Это коэффициент для величины CTE (Cross Track Error), то есть ошибки отклонения от центральной траектории
        # kepsi: 2.0  # Этот коэффициент является значением угловой ошибки отклонения от центральной траектории 
        # kev: 0.01 # Допустимая ошибка по скорости, то есть различее между желаемой скоростью и реальной
        # ksteer_cost: 5.0 # Это коэффициент стоимости угла поворота рулевого колеса. Он нужен, чтобы не было резких